apps/rush/src/utilities/Telemetry.ts

```diff
@@ -45,19 +47,55 @@ export default class Telemetry {
   }
 
   public flush(writeFile: (file: string, data: string) => void = fsx.writeFileSync): void {
-    if (!this._enabled) {
+    if (!this._enabled || this._store.length === 0) {
       return;
     }
+
     const fullPath: string = this._getFilePath();
     fsx.ensureDirSync(this._dataFolder);
     writeFile(fullPath, JSON.stringify(this._store));
     this._store = [];
+    this._cleanUp();
   }
 
   public get store(): ITelemetryData[] {
     return this._store;
   }
 
+  /**
+   * When there are too many log files, delete the old ones.
```

Comment:
>When there are too many log files, delete the old ones. [](start = 5, length = 55)

why is this a problem?

Reply:
So that rush telemetry data won't take up all the disk space.

---
In reply to: [138707994](https://github.com/Microsoft/web-build-tools/pull/371#discussion_r138707994) [](ancestors = 138707994)